api/src/main/java/org/openmrs/module/reporting/config/ReportLoader.java

```diff
@@ -203,47 +208,29 @@ public static void addDataSetDefinition(ReportDefinition rd, DataSetDescriptor d
 
         // Next, if any data set parameters specify values for report parameters, or are new parameters, add these
         List<Parameter> datasetParameters = constructParameters(dataSetDescriptor.getParameters());
-        if (datasetParameters != null) {
-            for (Parameter parameter : datasetParameters) {
-                boolean found = false;
-                for (Parameter existingParam : rd.getParameters()) {
-                    if (existingParam.getName().equals(parameter.getName())) {
-                        found = true;
-                        if (parameter.getDefaultValue() != null) {
-                            mappedDsd.getParameterMappings().put(existingParam.getName(), parameter.getDefaultValue());
-                            parameter.setDefaultValue(null);
-                        }
-                    }
-                }
-                if (!found) {
-                    mappedDsd.getParameterizable().addParameter(parameter);
+        for (Parameter parameter : datasetParameters) {
+            boolean found = false;
+            for (Parameter existingParam : rd.getParameters()) {
+                if (existingParam.getName().equals(parameter.getName())) {
+                    found = true;
                     if (parameter.getDefaultValue() != null) {
-                        mappedDsd.addParameterMapping(parameter.getName(), parameter.getDefaultValue());
+                        mappedDsd.getParameterMappings().put(existingParam.getName(), parameter.getDefaultValue());
                         parameter.setDefaultValue(null);
                     }
                 }
             }
+            if (!found) {
+                mappedDsd.getParameterizable().addParameter(parameter);
+                if (parameter.getDefaultValue() != null) {
+                    mappedDsd.addParameterMapping(parameter.getName(), parameter.getDefaultValue());
+                    parameter.setDefaultValue(null);
+                }
+            }
```

Comment:
Weird that it found a different here, I don't think there is any?

Reply:
No, that's odd